sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/IEDAdapterTest.java

```diff
@@ -172,91 +175,195 @@ void testMatches() throws Exception {
         assertTrue(iAdapter.matches(objectReference));
     }
 
-
     @Test
-    void testCreateDataSet() throws Exception {
+    void createDataSet_should_create_when_DataSetInfo_is_complete() throws Exception {
+        // Given
         SCL scd = SclTestMarshaller.getSCLFromFile("/ied-test-schema-conf/ied_unit_test.xml");
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
         IEDAdapter iAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"));
 
-        assertThrows(ScdException.class, () -> iAdapter.createDataSet(new DataSetInfo()));
-
-        TServices tServices = new TServices();
-        iAdapter.getCurrentElem().setServices(tServices);
-        assertFalse(iAdapter.hasDataSetCreationCapability());
-
-        TLogSettings tLogSettings = new TLogSettings();
-        tServices.setLogSettings(tLogSettings);
-        tLogSettings.setDatSet(TServiceSettingsEnum.CONF);
-        assertTrue(iAdapter.hasDataSetCreationCapability());
-
         TDataSet tDataSet = new TDataSet();
         tDataSet.setName("dataset");
         TFCDA tfcda = new TFCDA();
         tfcda.setFc(TFCEnum.ST);
         tDataSet.getFCDA().add(tfcda);
         DataSetInfo dataSetInfo = DataSetInfo.from(tDataSet);
         dataSetInfo.setHolderIEDName("IED_NAME");
-        assertThrows(ScdException.class, () -> iAdapter.createDataSet(dataSetInfo));
         dataSetInfo.setHolderLDInst("LD_INS2");
         dataSetInfo.setHolderLnClass(TLLN0Enum.LLN_0.value());
-        assertDoesNotThrow(() -> iAdapter.createDataSet(dataSetInfo));
 
+        // When Then
+        assertDoesNotThrow(() -> iAdapter.createDataSet(dataSetInfo));
         assertIsMarshallable(scd);
     }
+    @Test
+    void createDataSet_should_throw_ScdException_when_DataSetInfo_is_incomplete() throws Exception {
+        // Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/ied-test-schema-conf/ied_unit_test.xml");
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        IEDAdapter iAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"));
```

Comment:
many asserts in this test, you should remove this one as if it throw an exception, the test will fail

Reply:
OK